### DISCUSSION TITLE
The problem is password hashing in the admin panel.

### DISCUSSION THREAD

**Discussion by wnesbv** [aminalaee/sqladmin#112] [Q&A]
https://github.com/wnesbv/starlette-auth-admin

**Comment by wnesbv** [comment]
"password hashing," I decided.. But it must be decided-by you-at will. Need hashing functionality
https://github.com/wnesbv/starlette-auth-admin/commit/b30162f81d2191fd1fe3cfeb9ae42412bae035e5

**Comment by dwreeves** [comment]
Authentication is an important feature to support, but the admin panel should probably support a more general approach to authentication, such as the ability to add middleware.

I would open a PR where you expose `starlette.Starlette`'s middleware kwarg during the creation of the admin instance, rather than adding a new dependency and an opinionated authentication approach that may not work with what users want.

Additionally, there is the `ModelAdmin.is_accessible()` method, which allows for custom hooks.

**Reply by aminalaee** [reply]
Even though I think the same can be achieved with the `is_accessible`, but I guess it's ok to accept middlewares for the Admin app 👍

**Comment by wnesbv** [comment]
I have not met a ready-made solution for authorization and autification starlette, so - "if anything, come in"https://github.com/wnesbv/starlette-auth-admin
